src/main/frontend/handler/file.cljs

```diff
@@ -158,53 +158,60 @@
-                       (let [add-helper
-                             (fn []
-                               (doall
-                                (map
-                                 (fn [[path content]]
-                                   (git-handler/git-add repo path update-status?))
-                                 files)))]
-                         (-> (p/all (add-helper))
-                             (p/then (fn [_]
-                                       (when git-add-cb
-                                         (git-add-cb))))
-                             (p/catch (fn [error]
-                                        (println "Git add failed:")
-                                        (js/console.error error)))))
-                       (ui-handler/re-render-file!)
-                       (when add-history?
-                         (history/add-history! repo files-tx)))]
-       (-> (p/all (doall (map write-file-f files)))
-           (p/then git-add-f)
-           (p/catch (fn [error]
-                      (println "Alter files failed:")
-                      (js/console.error error))))))))
+                              (-> (p/let [_ (fs/check-directory-permission! repo)]
+                                    (fs/write-file (util/get-repo-dir repo) path content original-content))
+                                  (p/catch (fn [error]
+                                             (log/error :write-file/failed {:path path
+                                                                            :content content
+                                                                            :error error}))))))
+             git-add-f (fn []
+                         (let [add-helper
+                               (fn []
+                                 (doall
+                                  (map
+                                   (fn [[path content]]
+                                     (git-handler/git-add repo path update-status?))
+                                   files)))]
+                           (-> (p/all (add-helper))
+                               (p/then (fn [_]
+                                         (when git-add-cb
+                                           (git-add-cb))))
+                               (p/catch (fn [error]
+                                          (println "Git add failed:")
+                                          (js/console.error error)))))
+                         (ui-handler/re-render-file!)
+                         (when add-history?
+                           (history/add-history! repo files-tx)))]
+         (-> (p/all (doall (map write-file-f files)))
+             (p/then (fn []
+                       (git-add-f)))
+             (p/catch (fn [error]
+                        (println "Alter files failed:")
+                        (js/console.error error)))
+             (p/finally (fn [_]
+                          (reset! writing? false)))))))))
```

Comment:
Might not always reach here

Reply:
The outside block is query data from the datascript, which might throw errors but it's rare, actually, it should just use the normal `let`.